test_parse_directory.py

```diff
@@ -100,20 +112,31 @@ def main():
     errors = 0
     for root, dirs, files in os.walk(directory):
         for file in files:
-            # Only attempt to parse Python files
-            if not file.endswith(".py"):
-                continue
-
             file_path = os.path.join(root, file)
 
+            should_exclude_file = any(
+                [
+                    True
+                    for pattern in excluded_files
+                    if PurePath(file_path).match(pattern)
+                ]
```

Comment:
`any()` works better if you pass a generator expression instead of a list comprehension (it'll stop at the first truthy item generated). Would this more readable if you wrote it as a classic for-statement containing an if-statement?

Finally, maybe lift the `PurePath()` call out of the loop? (Could even use use the `joinpath()` method on a `PurePath` instance created from `root`?)

Otherwise LG.